server/game/cards/08-MotC/AsahinaTakamori.js

```diff
@@ -0,0 +1,31 @@
+const DrawCard = require('../../drawcard.js');
+const { CardTypes, Durations, Players } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl');
+
+class AsahinaTakamori extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            title: 'Choose a character to pacify',
+            when: {
+                onCardPlayed: (event) => event.card.type === CardTypes.Character && event.card.isFaction('crane')
+            },
+            target: {
+                controller: Players.Opponent,
+                cardType: CardTypes.Character,
+                cardCondition: (card, context) => card.getCost() <= context.event.card.getCost(),
+                gameAction: AbilityDsl.actions.cardLastingEffect({
+                    duration: Durations.UntilEndOfRound,
+                    effect: [
+                        AbilityDsl.effects.cannotParticipateAsAttacker(),
+                        AbilityDsl.effects.cannotParticipateAsDefender()
```

Comment:
I think it should be declareAsAttacker here.

Reply:
Ah good call.  Fix incoming.

Reply:
Huh, I thought we had that as an effect already, but doesn't look like it.  I'll have to write a new one :)

Reply:
Nevermind, found it as a `cardCannot` effect